### ISSUE TITLE
Termux not starting properly (Process completed (signal4)

### ISSUE THREAD

**Issue by battlezone** [termux/termux-app#916]
I am trying to run Termix on an old ASUS Transformer TF101 but whenever I start it I get the message:

> [Process completed (signal4) - press Enter]

When I press enter, Termux just closes.

I am running KatKiss 034, Android 6.0.1
The device itself has a Tegra 2 cpu which lacks NEON instructions, in case that is relevant.

Any known fix for this?

**Comment by Grimler91** [comment]
Is this in a fresh install? If not, re-install termux (or open a [fail-safe session](https://wiki.termux.com/wiki/Recover_a_broken_environment#Launching_a_Failsafe_Session) and see if the problem lies in some config file)

A logcat might tell us more about the issue (if the problem does not lie in, for example, .bashrc)

**Comment by xeffyr** [comment]
> Tegra 2 cpu which lacks NEON instructions,
> If not, re-install termux (or open a fail-safe session and see if the problem lies in some config file)

@Grimler91 Re-installing won't help as Tegra 2 really don't have NEON instructions. Problem exists due to https://github.com/termux/termux-packages/blob/master/build-package.sh#L600.

To fix that, we have to recompile whole repository for ARM with `-mfpu=vfp` instead of `-mfpu=neon`.

**Comment by battlezone** [comment]
Is this something I can do myself?
I'm guessing I can't since these packages are pulled from the internet.

**Comment by xeffyr** [comment]
@battlezone No. Assume that you have to recompile everything from source using PC & docker image.

**Comment by xeffyr** [comment]
Current Termux & packages build requires support for NEON instructions so Tegra 2 support is dropped. I guess that this issue will remain unresolved as Tegra 2 is old...
But let's wait what @fornwall will say about this. Replacing `-mfpu=neon` with `-mfpu=vfp` should be safe, so question is only about recompiling all ARM packages.

**Comment by battlezone** [comment]
Fail-safe session is working though. So I'm guessing that compiling the packages without NEON instructions (if that is even possible) would be the solution for me. I am building them with `-mfpu=vfp` now and so far it seems to be going well. I am not sure how I would get them loaded in Termux when I'm done building though.

I completely understand that there is no support for Tegra 2. I am just trying to bring this tablet back to life. Since it has a physical keyboard it would be perfect for running Termux.

**Comment by xeffyr** [comment]
Archive with basic packages (`-mfpu=vfp`): [base-pkgs-vfp.tar.gz](https://github.com/termux/termux-app/files/2578498/base-pkgs-vfp.tar.gz)

**Comment by battlezone** [comment]
Thanks a lot! I saw you posted the link so I went ahead and installed it before I saw your edit with instructions.

I actually moved the archive to `/data/data/com.termux/files` but when extracting the archive I got permission errors, even when changing ownership of the archive to the Termux app user. Extracting it as root and changing ownership of all the files and folders back to the Termux user worked fine as well. 
This might be helpful to anyone else who ever tries to run Termux on an ancient Android device.

Again, thanks a lot for your help!
